### PR TITLE
Update reporter.ts to allow for multiple CaseIDs

### PR DESCRIPTION
**Pull request by spencerlavallesonos**
E.g. 

`it('C00001 C00002 should do two things', async function () {...`

### CODE REVIEW

**Review by christian-bromann — COMMENTED** [review]
Thanks for the patch, mind adding some unit tests for this?

**Comment by spencerlavallesonos** [comment]
@christian-bromann Been trying to mock up all the testStats and have run into a roadblock, namely with TestStat's `private _stringifyDiffObjs`. 

Could use some assistance. 

I have some mockable test JSONs but am getting tripped up on mocking that essential property.

**Comment by christian-bromann** [comment]
How can I assist?

**Comment by spencerlavallesonos** [comment]
@christian-bromann Looking for some insight on how best to mock the `testStats` object parameter for the various `onTest*` hooks. 

Currently having difficulty around some of the `@wdio/reporter.TestStats` keys & types. Do you have an example unit test that uses `TestStats` that I can reference?

File: src/reporter.ts [hunk 1]
```diff
@@ -135,7 +147,7 @@ export default class TestRailReporter extends WDIOReporter {
         if (this.#options.useCucumber && suiteStats.type === 'scenario') {
             let testId = '-1'
             if (this.#options.caseIdTagPrefix && suiteStats.tags) {
-                for (let i = 0; i < suiteStats.tags.length; i++) {
+                for (let i = 0  i < suiteStats.tags.length  i++) {
```
Comment (christian-bromann) [inline src/reporter.ts:150]:
```suggestion
                for (let i = 0; i < suiteStats.tags.length; i++) {
```

**Comment by spencerlavallesonos** [comment]
@christian-bromann Unit tests have been added, and updated README to illustrate usage.

**Comment by christian-bromann** [comment]
mhm .. not sure where it comes from, mind pushing a fix for it into this working branch?

**Comment by christian-bromann** [comment]
@spencerlavallesonos can we update the Node.js version within the pipeline to Node v18?

**Comment by spencerlavallesonos** [comment]
> @spencerlavallesonos can we update the Node.js version within the pipeline to Node v18?

@christian-bromann I've updated the node-version matrix to run from v`[18.x, 20.x]`

**Comment by christian-bromann** [comment]
This does the trick but unfortunately there seem to be 2 eslint issues now.

**Comment by spencerlavallesonos** [comment]
> This does the trick but unfortunately there seem to be 2 eslint issues now.

Yep yep, sorry for all the back and forth, I'll make it squeaky clean now. 🖖

**Review by christian-bromann — APPROVED** [review]
Great work 🎉  Thanks a lot!

**Pull request merged**